monai/transforms/compose.py

```diff
@@ -188,21 +115,164 @@ def execute_compose(
     for _transform in transforms[start:end]:
         if threading:
             _transform = deepcopy(_transform) if isinstance(_transform, ThreadUnsafe) else _transform
-        data = evaluate_with_overrides(
-            data,
+        data = apply_transform(
             _transform,
-            lazy_evaluation=lazy_evaluation,
+            data,
+            map_items,
+            unpack_items,
+            lazy=lazy,
+            lazy_strategy=lazy_strategy,
             overrides=overrides,
-            override_keys=override_keys,
-            verbose=verbose,
+            logger_name=logger_name,
         )
-        data = apply_transform(_transform, data, map_items, unpack_items, log_stats)
-    data = evaluate_with_overrides(
-        data, None, lazy_evaluation=lazy_evaluation, overrides=overrides, override_keys=override_keys, verbose=verbose
-    )
+    if isinstance(data, (list, tuple)) and map_items:
+        data = [apply_pending_transforms(d, None, overrides, logger_name=logger_name) for d in data]
+    data = apply_pending_transforms(data, None, overrides, logger_name=logger_name)
     return data
 
 
+class ComposeCompiler:
```

Comment:
As we've discussed today at the meeting I'd suggest something like `LazyPolicy`  given the current state of this class. If the intent is to evolve this to do more complicated work to "compile" in some manner we might want to put that into a new class anyway and keep something to do policy like this separate.